### PR TITLE
8269115: WebView paste event contains old data

### PR DESCRIPTION
**Pull request by jaybhaskar**
Issue: The DataObject uses m_availMimeTypes as Vector of strings, and appending mime types in pasteboard operation like setPlainText, Hence it will append duplicate mime types in m_availMimeTypes , in this case the length clipboardData.types would be wrong, and duplicates data would be copied to clipboard target.
Solution: Use ListHashSet data Structure instead of Vector for m_availMimeTypes.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8269115](https://bugs.openjdk.java.net/browse/JDK-8269115): WebView paste event contains old data

### Reviewers
 * [Ambarish Rapte](https://openjdk.java.net/census#arapte) (@arapte - **Reviewer**)
 * [Kevin Rushforth](https://openjdk.java.net/census#kcr) (@kevinrushforth - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jfx pull/729/head:pull/729` \
`$ git checkout pull/729`

Update a local copy of the PR: \
`$ git checkout pull/729` \
`$ git pull https://git.openjdk.java.net/jfx pull/729/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 729`

View PR using the GUI difftool: \
`$ git pr show -t 729`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jfx/pull/729.diff">https://git.openjdk.java.net/jfx/pull/729.diff</a>

</details>

### CODE REVIEW

**Review by arapte — CHANGES_REQUESTED** [review]
The fix itself looks good. Suggested few minor changes.
Regarding test:
Observed that the test does not fail without this fix. Please recheck.
Also, I would recommend to add new tests to test the behavior mentioned in JBS, and modify any pre-existing tests if they now become invalid with fix.

File: modules/javafx.web/src/main/native/Source/WebCore/platform/java/PasteboardJava.cpp [hunk 1]
```diff
@@ -237,6 +237,7 @@ std::unique_ptr<Pasteboard> Pasteboard::createForCopyAndPaste(std::unique_ptr<Pa
     static RefPtr<DataObjectJava> data = DataObjectJava::create();
     // TODO: setURL, setFiles, setData, setHtml (needs URL)
     data->setPlainText(jGetPlainText());
+    data->setData(DataObjectJava::mimeHTML(),jGetPlainText());
```
[the remark this answers is not in the document]
Comment (jaybhaskar) [reply modules/javafx.web/src/main/native/Source/WebCore/platform/java/PasteboardJava.cpp:240]: done

File: modules/javafx.web/src/test/java/test/javafx/scene/web/HTMLEditingTest.java [hunk 2]
```diff
@@ -50,20 +51,26 @@
...
+
+            assertEquals("Target onpaste data", getEngine().
+                    executeScript("pasteTarget.numOfMimeTypes").toString(), "undefined");
+
             String clipboardData = "Clipboard text";
             ClipboardContent content = new ClipboardContent();
             content.putString(clipboardData);
             Clipboard.getSystemClipboard().setContent(content);
 
+
```
[the remark this answers is not in the document]
Comment (jaybhaskar) [reply modules/javafx.web/src/test/java/test/javafx/scene/web/HTMLEditingTest.java:73]: done

**Comment by jaybhaskar** [comment]
The commit was before the latest webkit update , yes the test pass without fix, i will investigate and apply the required change

**Comment by jaybhaskar** [comment]
All , requested changes applied.

File: modules/javafx.web/src/test/java/test/javafx/scene/web/HTMLEditingTest.java [hunk 3]
```diff
@@ -72,13 +86,14 @@
                             false, PlatformUtil.isMac()));// Cmd+V (Mac)
 
             assertEquals("Source Default value",getEngine().
-                    executeScript("srcInput.defaultValue").toString(),
-                    defaultText);
-            assertEquals("Source clipboard onpaste data", getEngine().
-                    executeScript("srcInput.value").toString(), clipboardData + defaultText);
+                    executeScript("srcInput.defaultValue").toString(), defaultText);
+            assertEquals("Source clipboard onpaste data", getEngine().executeScript("srcInput.value").
+                    toString(), clipboardData + defaultText);
```
Comment (kevinrushforth) [inline modules/javafx.web/src/test/java/test/javafx/scene/web/HTMLEditingTest.java:91]: The expected and actual values are backwards; the first of the two objects being compared should be the expected value. I see that this is a preexisting bug in the test. I recommend fixing this in the new test (while leaving the existing test method as is).

File: modules/javafx.web/src/test/java/test/javafx/scene/web/HTMLEditingTest.java [hunk 4]
```diff
@@ -72,13 +86,14 @@
...
             assertEquals("Target onpaste data", getEngine().
-                    executeScript("pasteTarget.value").toString(),
-                    clipboardData);
+                     executeScript("pasteTarget.value").toString(), clipboardData);
+            assertEquals("Target onpaste data size", getEngine().
+                            executeScript("document.getElementById(\"clipboardData\").innerText").toString(),
+                    "2");
```
Comment (kevinrushforth) [inline modules/javafx.web/src/test/java/test/javafx/scene/web/HTMLEditingTest.java:96]: Same comment as above about expected and actual being backwards.

**Review by kevinrushforth — COMMENTED** [review]
The fix looks good to me, but the test still needs additional changes. Then new test should fail without your fix and pass with your fix (as of now it passes either with or without your fix, so it isn't actually testing the fix).

Also, I think you might have misunderstood Ambarish's comment about the test. Rather than modifying the existing `clipboardGetDataOnPaste ` test method, I recommend to leave it alone and add a new test method.

**Review by kevinrushforth — APPROVED** [review]
The new manual test looks good. It fails without the fix and passes with the fix.

**Review by arapte — APPROVED** [review]
looks good to me too.

**Pull request closed without merging**